<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>Dialog_Preferences</class>
 <widget class="QDialog" name="Dialog_Preferences">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>543</width>
    <height>396</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Preferences</string>
  </property>
  <property name="windowIcon">
   <iconset theme="configure">
    <normaloff/>
   </iconset>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout">
     <item>
      <widget class="QTableWidget" name="tableWidget">
       <property name="minimumSize">
        <size>
         <width>128</width>
         <height>0</height>
        </size>
       </property>
       <property name="maximumSize">
        <size>
         <width>144</width>
         <height>16777215</height>
        </size>
       </property>
       <property name="selectionMode">
        <enum>QAbstractItemView::SingleSelection</enum>
       </property>
       <property name="selectionBehavior">
        <enum>QAbstractItemView::SelectRows</enum>
       </property>
       <property name="iconSize">
        <size>
         <width>48</width>
         <height>48</height>
        </size>
       </property>
       <property name="textElideMode">
        <enum>Qt::ElideRight</enum>
       </property>
       <property name="showGrid">
        <bool>false</bool>
       </property>
       <property name="cornerButtonEnabled">
        <bool>false</bool>
       </property>
       <attribute name="horizontalHeaderVisible">
        <bool>false</bool>
       </attribute>
       <attribute name="horizontalHeaderDefaultSectionSize">
        <number>144</number>
       </attribute>
       <attribute name="horizontalHeaderHighlightSections">
        <bool>false</bool>
       </attribute>
       <attribute name="horizontalHeaderStretchLastSection">
        <bool>true</bool>
       </attribute>
       <attribute name="verticalHeaderVisible">
        <bool>false</bool>
       </attribute>
       <attribute name="verticalHeaderDefaultSectionSize">
        <number>80</number>
       </attribute>
       <attribute name="verticalHeaderHighlightSections">
        <bool>false</bool>
       </attribute>
       <row>
        <property name="text">
         <string/>
        </property>
       </row>
       <row>
        <property name="text">
         <string/>
        </property>
       </row>
       <row>
        <property name="text">
         <string/>
        </property>
       </row>
       <row>
        <property name="text">
         <string/>
        </property>
       </row>
       <column>
        <property name="text">
         <string/>
        </property>
       </column>
       <item row="0" column="0">
        <property name="text">
         <string>General</string>
        </property>
        <property name="icon">
         <iconset theme="preferences-system">
          <normaloff/>
         </iconset>
        </property>
        <property name="flags">
         <set>ItemIsSelectable|ItemIsEnabled</set>
        </property>
       </item>
       <item row="1" column="0">
        <property name="text">
         <string>Toolbars</string>
        </property>
        <property name="icon">
         <iconset theme="configure-toolbars">
          <normaloff/>
         </iconset>
        </property>
        <property name="flags">
         <set>ItemIsSelectable|ItemIsEnabled</set>
        </property>
       </item>
       <item row="2" column="0">
        <property name="text">
         <string>Performance</string>
        </property>
        <property name="icon">
         <iconset theme="preferences-system-performance">
          <normaloff/>
         </iconset>
        </property>
       </item>
       <item row="3" column="0">
        <property name="text">
         <string>Interface</string>
        </property>
        <property name="icon">
         <iconset theme="preferences-desktop-theme">
          <normaloff/>
         </iconset>
        </property>
       </item>
      </widget>
     </item>
     <item>
      <widget class="QStackedWidget" name="stackedWidget">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Expanding" vsizetype="Preferred">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="currentIndex">
        <number>3</number>
       </property>
       <widget class="QWidget" name="page_general">
        <layout class="QGridLayout" name="gridLayout_2">
         <item row="1" column="1" colspan="2">
          <widget class="QComboBox" name="combo_language"/>
         </item>
         <item row="3" column="0" colspan="3">
          <widget class="QGroupBox" name="group_save">
           <property name="title">
            <string>Save between sessions</string>
           </property>
           <layout class="QVBoxLayout" name="verticalLayout_2">
            <item>
             <widget class="QCheckBox" name="check_save_toolbars">
              <property name="toolTip">
               <string>Save the customized toolbar buttons</string>
              </property>
              <property name="text">
               <string>Toolbars</string>
              </property>
              <property name="checked">
               <bool>true</bool>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QCheckBox" name="check_save_geometry">
              <property name="toolTip">
               <string>Dialogs, window size and position</string>
              </property>
              <property name="text">
               <string>User Interface</string>
              </property>
              <property name="checked">
               <bool>true</bool>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QCheckBox" name="check_save_style">
              <property name="text">
               <string>Knot Style</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QCheckBox" name="check_save_grid">
              <property name="text">
               <string>Grid</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QPushButton" name="button_clear_settings">
              <property name="toolTip">
               <string>Clear configuration and restore defaults next time Knotter is launched</string>
              </property>
              <property name="text">
               <string>Clear  saved configuration</string>
              </property>
              <property name="icon">
               <iconset theme="edit-clear">
                <normaloff/>
               </iconset>
              </property>
             </widget>
            </item>
           </layout>
          </widget>
         </item>
         <item row="0" column="2">
          <widget class="QPushButton" name="button_clear_recent">
           <property name="toolTip">
            <string>Clear recent file history</string>
           </property>
           <property name="text">
            <string>Clear</string>
           </property>
           <property name="icon">
            <iconset theme="edit-clear-history">
             <normaloff/>
            </iconset>
           </property>
          </widget>
         </item>
         <item row="1" column="0">
          <widget class="QLabel" name="label_3">
           <property name="text">
            <string>Language</string>
           </property>
          </widget>
         </item>
         <item row="2" column="0" colspan="3">
          <widget class="QCheckBox" name="check_files_close">
           <property name="text">
            <string>Check for unsaved files on close</string>
           </property>
           <property name="checked">
            <bool>true</bool>
           </property>
          </widget>
         </item>
         <item row="0" column="0">
          <widget class="QLabel" name="label_4">
           <property name="text">
            <string>Max Recent Files</string>
           </property>
          </widget>
         </item>
         <item row="0" column="1">
          <widget class="QSpinBox" name="spin_recent_files">
           <property name="maximum">
            <number>128</number>
           </property>
           <property name="value">
            <number>5</number>
           </property>
          </widget>
         </item>
         <item row="4" column="0" colspan="3">
          <widget class="QGroupBox" name="groupBox_4">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="title">
            <string>Clipboard</string>
           </property>
           <layout class="QHBoxLayout" name="horizontalLayout_4">
            <item>
             <widget class="QCheckBox" name="check_clipboard_knot">
              <property name="enabled">
               <bool>false</bool>
              </property>
              <property name="sizePolicy">
               <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="text">
               <string>Knot</string>
              </property>
              <property name="checked">
               <bool>true</bool>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QCheckBox" name="check_clipboard_xml">
              <property name="text">
               <string>XML</string>
              </property>
              <property name="icon">
               <iconset theme="text-xml">
                <normaloff/>
               </iconset>
              </property>
              <property name="checked">
               <bool>true</bool>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QCheckBox" name="check_clipboard_svg">
              <property name="text">
               <string>SVG</string>
              </property>
              <property name="icon">
               <iconset theme="image-svg+xml">
                <normaloff/>
               </iconset>
              </property>
              <property name="checked">
               <bool>true</bool>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QCheckBox" name="check_clipboard_png">
              <property name="text">
               <string>PNG</string>
              </property>
              <property name="icon">
               <iconset theme="image-x-generic">
                <normaloff/>
               </iconset>
              </property>
              <property name="checked">
               <bool>true</bool>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QCheckBox" name="check_clipboard_tiff">
              <property name="text">
               <string>TIFF</string>
              </property>
              <property name="icon">
               <iconset theme="image-x-generic">
                <normaloff/>
               </iconset>
              </property>
             </widget>
            </item>
           </layout>
          </widget>
         </item>
        </layout>
       </widget>
       <widget class="QWidget" name="page_toolbars">
        <layout class="QGridLayout" name="gridLayout">
         <item row="0" column="0">
          <widget class="QLabel" name="label">
           <property name="text">
            <string>Icon Size</string>
           </property>
          </widget>
         </item>
         <item row="0" column="1">
          <widget class="QComboBox" name="combo_icon_size">
           <property name="toolTip">
            <string>Icon size</string>
           </property>
           <property name="whatsThis">
            <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Select the size used for toolbar icons and items in the drawing area.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
           </property>
          </widget>
         </item>
         <item row="1" column="0">
          <widget class="QLabel" name="label_2">
           <property name="text">
            <string>Toolbar Button Style</string>
           </property>
          </widget>
         </item>
         <item row="1" column="1">
          <widget class="QComboBox" name="combo_icon_style">
           <property name="whatsThis">
            <string>Select how the toolbar should display icons and text</string>
           </property>
          </widget>
         </item>
         <item row="2" column="0" colspan="2">
          <widget class="QGroupBox" name="groupBox">
           <property name="title">
            <string>Toolbar Buttons</string>
           </property>
           <layout class="QHBoxLayout" name="horizontalLayout_2">
            <item>
             <widget class="Toolbar_Editor" name="toolbar_editor"/>
            </item>
           </layout>
          </widget>
         </item>
        </layout>
       </widget>
       <widget class="QWidget" name="page">
        <layout class="QVBoxLayout" name="verticalLayout_4">
         <item>
          <widget class="QGroupBox" name="groupBox_3">
           <property name="title">
            <string>Rendering</string>
           </property>
           <layout class="QVBoxLayout" name="verticalLayout_3">
            <item>
             <widget class="QCheckBox" name="check_cache_image">
              <property name="toolTip">
               <string>Enable caching of the rendered knot image</string>
              </property>
              <property name="whatsThis">
               <string>Enable caching of the rendered knot image. Imporves performance in large scenes but may create some artifacts.</string>
              </property>
              <property name="text">
               <string>Cache image</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QCheckBox" name="check_fluid_refresh">
              <property name="toolTip">
               <string>Redraw the knot while moving nodes</string>
              </property>
              <property name="whatsThis">
               <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;When selected the knot will be updated fluidly when moving nodes.&lt;/p&gt;&lt;p&gt;When disabled, will update only after the nodes have been moved.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
              </property>
              <property name="text">
               <string>Fluid refresh</string>
              </property>
              <property name="checked">
               <bool>true</bool>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QCheckBox" name="check_antialiasing">
              <property name="text">
               <string>Antialiasing</string>
              </property>
             </widget>
            </item>
           </layout>
          </widget>
         </item>
         <item>
          <widget class="QGroupBox" name="groupBox_2">
           <property name="title">
            <string>Script</string>
           </property>
           <layout class="QHBoxLayout" name="horizontalLayout_3">
            <item>
             <widget class="QLabel" name="label_5">
              <property name="text">
               <string>Timeout</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QSpinBox" name="spin_timeout">
              <property name="toolTip">
               <string>Maximum execution time for scripts in seconds, 0 means no limit.</string>
              </property>
              <property name="suffix">
               <string>s</string>
              </property>
              <property name="maximum">
               <number>60</number>
              </property>
             </widget>
            </item>
           </layout>
          </widget>
         </item>
         <item>
          <spacer name="verticalSpacer">
           <property name="orientation">
            <enum>Qt::Vertical</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>20</width>
             <height>40</height>
            </size>
           </property>
          </spacer>
         </item>
        </layout>
       </widget>
       <widget class="QWidget" name="page_2">
        <layout class="QGridLayout" name="gridLayout_3">
         <item row="0" column="0">
          <widget class="QLabel" name="label_6">
           <property name="text">
            <string>Node Icon Size</string>
           </property>
          </widget>
         </item>
         <item row="1" column="0">
          <widget class="QLabel" name="label_7">
           <property name="text">
            <string>Widget Style</string>
           </property>
          </widget>
         </item>
         <item row="0" column="1">
          <widget class="QComboBox" name="combo_node_size"/>
         </item>
         <item row="1" column="1">
          <widget class="QComboBox" name="combo_widget_style">
           <property name="toolTip">
            <string>User interface style. Some widgets may require a restart to take effect.</string>
           </property>
          </widget>
         </item>
         <item row="2" column="0" colspan="2">
          <widget class="QGroupBox" name="group_style_preview">
           <property name="toolTip">
            <string>Shows a preview of the widget style</string>
           </property>
           <property name="title">
            <string>Preview</string>
           </property>
           <layout class="QHBoxLayout" name="horizontalLayout_5">
            <item>
             <widget class="QPushButton" name="pushButton">
              <property name="text">
               <string>Button</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QSpinBox" name="spinBox"/>
            </item>
            <item>
             <widget class="QComboBox" name="comboBox">
              <item>
               <property name="text">
                <string>Item1</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string>Item2</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string>Item3</string>
               </property>
              </item>
             </widget>
            </item>
           </layout>
          </widget>
         </item>
         <item row="3" column="0" colspan="2">
          <widget class="QGroupBox" name="groupBox_5">
           <property name="title">
            <string>Colors</string>
           </property>
           <layout class="QGridLayout" name="gridLayout_4">
            <item row="0" column="0">
             <widget class="QLabel" name="label_8">
              <property name="text">
               <string>Grid</string>
              </property>
             </widget>
            </item>
            <item row="0" column="1">
             <widget class="Color_Selector" name="color_grid_lines"/>
            </item>
            <item row="1" column="0">
             <widget class="QLabel" name="label_9">
              <property name="text">
               <string>Node (Resting)</string>
              </property>
             </widget>
            </item>
            <item row="1" column="1">
             <widget class="Color_Selector" name="color_node_resting"/>
            </item>
            <item row="2" column="0">
             <widget class="QLabel" name="label_10">
              <property name="text">
               <string>Node (Highlighted)</string>
              </property>
             </widget>
            </item>
            <item row="2" column="1">
             <widget class="Color_Selector" name="color_node_highlighted"/>
            </item>
            <item row="3" column="0">
             <widget class="QLabel" name="label_11">
              <property name="text">
               <string>Edge (Resting)</string>
              </property>
             </widget>
            </item>
            <item row="3" column="1">
             <widget class="Color_Selector" name="color_edge_resting"/>
            </item>
            <item row="4" column="0">
             <widget class="QLabel" name="label_12">
              <property name="text">
               <string>Edge (Highlighted)</string>
              </property>
             </widget>
            </item>
            <item row="4" column="1">
             <widget class="Color_Selector" name="color_edge_highlighted"/>
            </item>
           </layout>
          </widget>
         </item>
        </layout>
       </widget>
      </widget>
     </item>
    </layout>
   </item>
   <item>
    <widget class="QDialogButtonBox" name="buttonBox">
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
     <property name="standardButtons">
      <set>QDialogButtonBox::Cancel|QDialogButtonBox::Ok</set>
     </property>
    </widget>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>Color_Preview</class>
   <extends>QWidget</extends>
   <header>color_preview.hpp</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>Color_Selector</class>
   <extends>Color_Preview</extends>
   <header>color_selector.hpp</header>
  </customwidget>
  <customwidget>
   <class>Toolbar_Editor</class>
   <extends>QWidget</extends>
   <header>toolbar_editor.hpp</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>buttonBox</sender>
   <signal>accepted()</signal>
   <receiver>Dialog_Preferences</receiver>
   <slot>accept()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>247</x>
     <y>355</y>
    </hint>
    <hint type="destinationlabel">
     <x>157</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>buttonBox</sender>
   <signal>rejected()</signal>
   <receiver>Dialog_Preferences</receiver>
   <slot>reject()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>315</x>
     <y>355</y>
    </hint>
    <hint type="destinationlabel">
     <x>286</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>tableWidget</sender>
   <signal>cellClicked(int,int)</signal>
   <receiver>stackedWidget</receiver>
   <slot>setCurrentIndex(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>98</x>
     <y>75</y>
    </hint>
    <hint type="destinationlabel">
     <x>220</x>
     <y>18</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>Dialog_Preferences</sender>
   <signal>accepted()</signal>
   <receiver>Dialog_Preferences</receiver>
   <slot>set_preferences()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>536</x>
     <y>46</y>
    </hint>
    <hint type="destinationlabel">
     <x>539</x>
     <y>94</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>set_preferences()</slot>
 </slots>
</ui>
